src/stat.rs

```diff
@@ -38,30 +50,48 @@ pub struct StatOptions {
 
 /// Run perf stat on the given command and event combinations. Currently starts and stops a cycles timer in serial for each event specified.
 pub fn run_stat(options: &StatOptions) {
-    println!("{:?}:\n {:?}", options.command, options.event);
     //demonstrating from cli. In future rather than starting and stopping counter in series for each event, events will have the ability to be added in groups that will coordinate their timing.
+    struct EventCounter {
+        event: Event,
+        start: isize,
+        stop: isize,
+    }
+
+    let mut event_list: Vec<EventCounter> = Vec::new();
+    let mut child = Command::new(&options.command[0])
+        .args(&options.command[1..])
+        .spawn()
+        .unwrap();
+    //prevent race condition on child program run time on most programs
+    unsafe { kill(child.id() as i32, libc::SIGSTOP) };
```

Comment:
Wouldn't this have the effect of missing events for the process that executes the command? The command is spawned, but then you tell it to stop immediately after.